WordPress/gutenberg
# Global Styles: Use context when translating entries in theme.json

This PR attaches a translation context to the strings defined in `experimental-theme.json`, following the conversations at

- `block.json` – https://github.com/WordPress/wordpress-develop/pull/868#discussion_r556528330
- `theme.json` – https://github.com/WordPress/gutenberg/pull/27380#issuecomment-753879323

> 1. WP-CLI adds context to all the extracted strings (see https://github.com/wp-cli/i18n-command/blob/0fa24efa7460564d17fa4c6e1763e806eb1613c3/src/BlockExtractor.php#L43-L69 how it's done for `block.json` already
> 2. The code in WordPress that actually loads the `theme.json` (and also `block.json`) file will run all strings through `translate_with_gettext_context()` with the same context.

It uses an auxiliary structure defined https://github.com/WordPress/gutenberg/pull/28246#issuecomment-761117898 to know which data should be translated.

## How to test

- Verify that the PHP test passes.
- Install and activate the TT1-blocks theme and load the site editor. Verify that presets are working as expected and there're no errors in the logs.
